crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -2170,9 +2170,8 @@ where
             ExprKind::Subscript { value, slice, .. } => {
                 // Ex) Optional[...], Union[...]
                 if self.ctx.in_type_definition
-                    && !self.ctx.in_deferred_string_type_definition
                     && !self.settings.pyupgrade.keep_runtime_typing
-                    && self.settings.rules.enabled(Rule::TypingUnion)
+                    && self.settings.rules.enabled(Rule::NonPEP604Annotation)
```

Comment:
My understanding is that our guideline is to use rule names that work well with `allow`. Is this an internal-only name and what's the motivation of using the PEP names?

Reply:
python features often lack a name beyond the PEP number. e.g. https://docs.python.org/3/library/typing.html says:

> [PEP 604](https://peps.python.org/pep-0604/): Allow writing union types as X | Y
> Introducing [types.UnionType](https://docs.python.org/3/library/types.html#types.UnionType) and the ability to use the binary-or operator | to signify a [union of types](https://docs.python.org/3/library/stdtypes.html#types-union)

we could invent our own name such as `NonPipeUnion`.